pkg/components/prometheus-operator/component_conversion_test.go

```diff
@@ -0,0 +1,171 @@
+		component "prometheus-operator" {
+		  prometheus {
+		    ingress {
+		      host                       = "prometheus.mydomain.net"
+		      class                      = "contour"
+		      certmanager_cluster_issuer = "letsencrypt-production"
+		    }
+		  }
+		}
+		`,
+			ExpectedConfigFileName: "prometheus-operator/templates/prometheus/prometheus.yaml",
+			Expected:               reflect.ValueOf("https://prometheus.mydomain.net"),
+			JSONPath:               "{.spec.externalUrl}",
+		},
+		{
+			Name: "ingress creation for prometheus",
+			InputConfig: `
+		component "prometheus-operator" {
+		  prometheus {
+		    ingress {
+		      host                       = "prometheus.mydomain.net"
+		      class                      = "contour"
+		      certmanager_cluster_issuer = "letsencrypt-production"
+		    }
+		  }
+		}
+		`,
+			ExpectedConfigFileName: "prometheus-operator/templates/prometheus/ingress.yaml",
+			Expected:               reflect.ValueOf("prometheus.mydomain.net"),
+			JSONPath:               "{.spec.rules[0].host}",
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+		t.Run(tc.Name, func(t *testing.T) {
+			t.Parallel()
+
+			m := renderManifests(t, tc.InputConfig)
+			if len(m) == 0 {
+				t.Fatalf("Rendered manifests shouldn't be empty")
+			}
+
+			gotConfig, ok := m[tc.ExpectedConfigFileName]
+			if !ok {
+				t.Fatalf("Config not found with filename: %q", tc.ExpectedConfigFileName)
+			}
+
+			u := getUnstructredObj(t, gotConfig)
+			got := getValFromObject(t, tc.JSONPath, u)
+
+			switch got.Kind() { //nolint:exhaustive
+			case reflect.Interface:
+				switch gotVal := got.Interface().(type) { //nolint:gocritic
+				case string:
+					expVal, ok := tc.Expected.Interface().(string)
+					if !ok {
+						t.Fatalf("expected value is not string")
+					}
+
+					if gotVal != expVal {
+						t.Fatalf("expected: %s, got: %s", expVal, gotVal)
+					}
+
+					// Add more cases as the expected values become heterogeneous.
+					// case bool:
+				}
+			default:
+				t.Log(got.Kind())
+			}
+		})
+	}
+}
+
+func getUnstructredObj(t *testing.T, yamlObj string) *unstructured.Unstructured {
+	u := &unstructured.Unstructured{}
+
+	// decode YAML into unstructured.Unstructured
+	dec := yamlserializer.NewDecodingSerializer(unstructured.UnstructuredJSONScheme)
+	if _, _, err := dec.Decode([]byte(yamlObj), nil, u); err != nil {
+		t.Fatalf("Converting config to unstructured.Unstructured: %v", err)
+	}
+
+	return u
+}
+
+func getValFromObject(t *testing.T, jp string, obj *unstructured.Unstructured) reflect.Value {
+	jPath := jsonpath.New("parse")
+	if err := jPath.Parse(jp); err != nil {
+		t.Fatalf("Parsing JSONPath: %v", err)
+	}
+
+	v, err := jPath.FindResults(obj.Object)
+	if err != nil {
+		t.Fatalf("Finding results using JSONPath in the YAML file: %v", err)
+	}
+
+	var got reflect.Value
+
+	if len(v) == 0 { //nolint:gocritic
+		t.Fatalf("No result found")
+	} else if len(v[0]) == 0 {
+		t.Fatalf("No result found")
```

Comment:
Can be simplified with `||` (as 2nd expression won't get evaluated if first one is true.
```suggestion
	if len(v) == 0 { //nolint:gocritic
		t.Fatalf("No result found")
	} else if len(v[0]) == 0 {
		t.Fatalf("No result found")
```